sections/docker.zsh

```diff
@@ -22,13 +22,12 @@ SPACESHIP_DOCKER_COLOR="${SPACESHIP_DOCKER_COLOR="cyan"}"
 spaceship_docker() {
   [[ $SPACESHIP_DOCKER_SHOW == false ]] && return
 
-  spaceship::exists docker || return
-
   # Show Docker status only for Docker-specific folders
   [[ -f $COMPOSE_FILE || -f Dockerfile || -f docker-compose.yml ]] || return
 
+  spaceship::exists docker || return
```

Comment:
Curious how this affects the performance ?

Reply:
If docker-specific folders or files do not exist, it should `return` in advance. So I put `paceship::exists docker || return` behind. @salmanulfarzy

Reply:
No.  It is obvious.

\\ | command exists & file exist | command exists & no file exist | no command exists & file exist | no command exists & no file exist
---|---|---|---|---
my version|2|1|2|1
current version|2|2|1|1

> 1 => Either command checking or file checking
> 2 => Both command checking and file checking

Mostly, if we enable `docker` section, `docker` exists in our terminal env.
So the following one left:

\\ | command exists & file exist | command exists & no file exist
---|---|---
my version|2|1
current version|2|2

Mostly, our terminal working directories do not have to do with docker.

So the following one left:

\\ |command exists & no file exist
---|---
my version|1
current version|2

Reply:
> Mostly, if we enable docker section, docker exists in our terminal env.

`docker` section is enabled by default and most users may not have `docker` installed. With the current implementation we don't have to check for files if the user doesn't have `docker` installed (case 3 on your table).

Reply:
> case 3: no command exists & file exist

If docker-specific folders/files exist, most users mostly works on docker-based project. Users will install it in the end.
I am a docker user, so I use this docker section. As you said, if most users may not have docker installed, we should disable it by default.

To docker users, `docker` is enabled. We should check docker-specific folders/files firstly.

Reply:
Except this tweak, are others OK? @salmanulfarzy

Reply:
It goes both ways, We don't have to check for files, if related `command` is unavailable. 

> are others OK?

Yes, looks good. May be an option to show complete `docker` version ?

Reply:
Currently, I find many sections do not support show complete version. To keep consistent, I think it is not time to support it now. Your opinion ? @salmanulfarzy

Reply:
Mentioned because existing users might expect to see the full version. We could leave that if that information is not important.

Also do you mind moving the `npm test` improvement to new PR, To keep changes separate ? Don't want to mix things together with Squash and Merge.

Reply:
@salmanulfarzy  OK. I reverted it to just keep `npm test` PR.

Reply:
Don't forget to submit new PR with proposed `docker` modifications.